shim/src/proxy.js

```diff
@@ -0,0 +1,68 @@
+import {
+  defineProperty,
+  deleteProperty,
+  getOwnPropertyDescriptor,
+  getOwnPropertyNames
+} from './commons';
+
+// this flag allow us to determine if the eval() call is a controlled eval done by the realm's code
+// or if it is user-land invocation, so we can react differently.
+let isInternalEvaluation = false;
+
+export function setInternalEvaluation() {
```

Comment:
isn't it dangerous to export these publicly?

Reply:
Original code. Removed.